src/js/media-error.js

```diff
@@ -75,6 +76,15 @@ MediaError.prototype.message = '';
  */
 MediaError.prototype.status = null;
 
+/**
+ * An optional error type string to give more detail about the error. This can be used to give
+ * a higher level of specificity to an error versus the more generic MediaError codes.
+ * This value should align with the values from videojs.Errors
+ *
+ * @type {string}
+ */
+MediaError.prototype.errorType = null;
```

Comment:
I would suggest something more robust,

interface VideoJsError {
  errorType: ErrorType
}

interface ErrorMetadata {
  error: VideoJsError // other errors may have additional properties other than just errorType
}

MediaError.metadata: ErrorMetadata

Reply:
Good call, it would probably be nice to pass more metadata than just the type if we want. The issue here is that `interface` is not allowed in JS.

We could change `errorType` to `metadata` here, and then just allow any properties to be tied to it. I am trying to think of a way in JS to be more strict about what `metadata` could contain though, without using TS.

Reply:
You can use typescipt interfaces using typedef in jsdoc

Reply:
Makes sense, I had to toggle with it a bit, but I made some updates to the JSDoc to ensure that is all working. Let me know if it looks correct to you.